tests/test_validation_pandapipes.py

```diff
@@ -0,0 +1,268 @@
+
+                Returns
+                -------
+                The timeseries
+                """
+                return super().times(variable)[:9]
+
+        demand_time_series_file = "timeseries_constant.csv"
+
+        solution = run_esdl_mesido_optimization(
+            SourcePipeSinkNetwork,
+            base_folder=base_folder,
+            esdl_file_name="Test_Tree_S1C1.esdl",
+            esdl_parser=ESDLFileParser,
+            profile_reader=ProfileReaderFromFile,
+            input_timeseries_file=demand_time_series_file,
+        )
+        results = solution.extract_results()
+
+        demand_matching_test(solution, results)
+        energy_conservation_test(solution, results)
+        heat_to_discharge_test(solution, results)
+
+        # ------------------------------------------------------------------------------
+        # ------------------------------------------------------------------------------
+        # Setup and run pandapipes
+        # ------------------------------------------------------------------------------
+        # ------------------------------------------------------------------------------
+        from examples.pandapipes.src.pandapipeesdlparser import PandapipeEsdlParserClass
+
+        esdl_file = os.path.join(base_folder, "model", "Test_Tree_S1C1.esdl")
+        esdlparser = PandapipeEsdlParserClass()
+        esdlparser.loadESDLFile(esdl_file)
+        esdlparser.add_additional_joint()
+
+        # ------------------------------------------------------------------------------
+        # ------------------------------------------------------------------------------
+        # Setup scenario
+        total_producers = len(esdlparser.esdl_asset["heat"]["producer"])
+        total_consumers = len(esdlparser.esdl_asset["heat"]["consumer"])
+
+        # Create panda_pipes network
+        net, net_asset, supply_temperature, return_temperature = esdlparser.createpandapipenet()
+
+        # Setup profile data
+        raw_profile_demand_load_watt = pd.read_csv(
+            os.path.join(base_folder, "input", demand_time_series_file)
+        )
+        profile_demand_load_watt = pd.DataFrame(columns=["0"])
+        profile_demand_load_watt["0"] = raw_profile_demand_load_watt["demand_1"]
+
+        demand_power = profile_demand_load_watt["0"].to_numpy() / total_consumers
+        profile_demand_load_watt = pd.DataFrame(results["demand_1.Heat_demand"])
+
+        # Setup supply mass flow for panda_pipes
+        average_temperature_kelvin = (supply_temperature + return_temperature) / 2.0 + 273.15
+        cp_joule_kgkelvin = pp.get_fluid(net).get_heat_capacity(average_temperature_kelvin)
+
+        # Enforce mass flow rate instead of cacluting it from Q = m_dot...
+        mesido_demand_flow_kg_s = results["Pipe1.Q"] * 988.0
+        demand_flow = mesido_demand_flow_kg_s
+        supply_flow = demand_flow * total_consumers / total_producers
+
+        # Assign profiles to assets
```

Comment:
If I understand correctly the constControl is basically setting the boundary conditions. Can we in a comment state which boundary conditions we set and what states now follow from the simulation.

Reply:
Comment has been added and imrpoved